Sources/Spezi/Dependencies/Module+Dependencies.swift

```diff
@@ -8,52 +8,66 @@
     /// ```swift
     /// class ExampleModule: Module {
-    ///     @Dependency var exampleModuleDependency: ExampleModuleDependency
+    ///     @Dependency var exampleDependency: ExampleModuleDependency
     /// }
     /// ```
-    public typealias Dependency<M: Module> = _DependencyPropertyWrapper<M>
-    
-    
-    /// Defines dynamic dependencies to other ``Module``s.
     ///
-    /// In contrast to the `@Dependency` property wrapper, the `@DynamicDependencies` enables the generation of the property wrapper in the initializer and generating an
-    /// arbitrary amount of dependencies that are resolved in the Spezi initialization.
+    /// ### Optional Dependencies
+    ///
+    /// You can define dependencies to be optional by declaring the `@Dependency` property wrapper optional.
+    /// The below code examples demonstrates this functionality.
+    ///
+    /// ```swift
+    /// class ExampleModule: Module {
+    ///     @Dependency var exampleDependency: ExampleModuleDependency?
+    ///
+    ///     func configure() {
+    ///         if let exampleDependency {
+    ///             // Dependency was defined by the user ...
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Computed Dependencies
+    ///
+    /// In certain circumstances your list of dependencies might not be statically known. Instead you might want to generate
+    /// your list of dependencies within the initializer, based on external factors.
+    /// The `@Dependency` property wrapper, allows you to define your dependencies using a result-builder-based appraoch.
     ///
-    /// - Note: You can access the contents of `@DynamicDependencies` once your ``Module/configure()-5pa83`` method is called (e.g., it must not be used in the `init`)
-    ///     and can continue to access the Standard actor in methods like ``LifecycleHandler/willFinishLaunchingWithOptions(_:launchOptions:)-8jatp``.
+    /// Below is a short code example that demonstrates this functionality.
     ///
-    /// A ``Module`` can define dynamic dependencies using the @``Module/DynamicDependencies`` property wrapper and can, e.g., initialize its value in the initializer.
     /// ```swift
     /// class ExampleModule: Module {
-    ///     @DynamicDependencies var dynamicDependencies: [any Module]
+    ///     @Dependency var computedDependencies: [any Module]
     ///
     ///
     ///     init() {
-    ///         self._dynamicDependencies = DynamicDependencies(
-    ///             moduleProperty: [
-    ///                 Dependency(wrappedValue: /* ... */),
-    ///                 Dependency(wrappedValue: /* ... */)
-    ///             ]
-    ///         )
+    ///         // a result builder to declare your module dependencies
+    ///         self._computedDependencies = Dependency {
+    ///             ModuleA()
+    ///             ModuleB()
+    ///         }
```

Comment:
Nice, this looks really great, good to see that everything is now a `Dependency`.